pkg/pool-weighted/contracts/lbp/FixedPriceLBPool.sol

```diff
@@ -0,0 +1,336 @@
+    }
+
+    /**
+     * @notice For bi-directional (buy and sell) sales, this is the maximum deviation from proportional initialization.
+     * @return initializationTolerancePercentage The maximum deviation from proportionality allowed at init time
+     */
+    function getInitializationTolerancePercentage() external pure returns (uint256) {
+        return _INITIALIZATION_TOLERANCE_PERCENTAGE;
+    }
+
+    /// @inheritdoc IFixedPriceLBPool
+    function getProjectTokenRate() external view returns (uint256) {
+        return _projectTokenRate;
+    }
+
+    /// @inheritdoc IFixedPriceLBPool
+    function getFixedPriceLBPoolDynamicData() external view returns (FixedPriceLBPoolDynamicData memory data) {
+        data.balancesLiveScaled18 = _vault.getCurrentLiveBalances(address(this));
+        data.staticSwapFeePercentage = _vault.getStaticSwapFeePercentage((address(this)));
+        data.totalSupply = totalSupply();
+
+        PoolConfig memory poolConfig = _vault.getPoolConfig(address(this));
+        data.isPoolInitialized = poolConfig.isPoolInitialized;
+        data.isPoolPaused = poolConfig.isPoolPaused;
+        data.isPoolInRecoveryMode = poolConfig.isPoolInRecoveryMode;
+        data.isSwapEnabled = _isSwapEnabled();
+    }
+
+    /// @inheritdoc IFixedPriceLBPool
+    function getFixedPriceLBPoolImmutableData() external view returns (FixedPriceLBPoolImmutableData memory data) {
+        data.tokens = _vault.getPoolTokens(address(this));
+        data.projectTokenIndex = _projectTokenIndex;
+        data.reserveTokenIndex = _reserveTokenIndex;
+
+        (data.decimalScalingFactors, ) = _vault.getPoolTokenRates(address(this));
+        data.isProjectTokenSwapInBlocked = _blockProjectTokenSwapsIn;
+        data.startTime = _startTime;
+        data.endTime = _endTime;
+
+        data.projectTokenRate = _projectTokenRate;
+
+        // Migration-related params, non-zero if the pool supports migration.
+        data.migrationRouter = _migrationRouter;
+        data.lockDurationAfterMigration = _lockDurationAfterMigration;
+        data.bptPercentageToMigrate = _bptPercentageToMigrate;
+        data.migrationWeightProjectToken = _migrationWeightProjectToken;
+        data.migrationWeightReserveToken = _migrationWeightReserveToken;
+    }
+
+    /*******************************************************************************
+                                    Base Pool Hooks
+    *******************************************************************************/
+
+    /// @inheritdoc IBasePool
+    function onSwap(PoolSwapParams memory request) public view override returns (uint256 amountCalculatedScaled18) {
+        // Block if the sale has not started or has ended.
+        if (_isSwapEnabled() == false) {
+            revert SwapsDisabled();
+        }
+
+        // If project token swaps are blocked, project token must be the token out.
+        if (_blockProjectTokenSwapsIn && request.indexOut != _projectTokenIndex) {
```

Comment:
```suggestion
        if (_blockProjectTokenSwapsIn && request.indexIn == _projectTokenIndex) {
```
if project token in is blocked and token in is project, revert --> this is probably more consistent from a readability standpoint, as the condition matches the flag.

Reply:
(actually see comment in the lines below)